### PR TITLE
News: Release 0.12

### PR DESCRIPTION
**Pull request by cart**
## How this works

(This contains sections as of <https://github.com/bevyengine/bevy/commit/a042924a6ee520a0b0dc72bc287a6f5f5f305002> ... still plenty of open things to merge!)

For the Bevy 0.12 release blog post, other people (besides me) can write blog post sections. Specifically, if you worked on a feature in a substantial way and are interested in presenting it, you can now ask to claim a section by leaving a comment in this PR. If you claim a section, submit a pull request to the `release-0.12.0` branch in this repo. For the next week, we will be filling in sections (the release target is Friday October 20th or Saturday October 21st). Please don't claim a section if you don't plan on completing it within that timeline. Also don't claim a section if you weren't an active participant in the design and implementation of the change (unless you are a Maintainer or SME).

I will claim any unclaimed sections.

Try to match the style of previous release blog posts as much as possible.

1. Show, don't tell. Don't bombard people with information. Avoid large walls of text _and_ large walls of code. Prefer the pattern "byte sized description of one thing" -> "example code/picture/video contextualizing that one thing" -> repeat. Take readers on a journey step by simple step.
2. Don't use up reader's "mental bandwidth" without good reason. We can't afford page-long descriptions of minor bug fixes. If it isn't a "headliner change", keep the description short and sweet. If a change is self describing, let it do that (ex: We now support this new mesh shape primitive ... this is what it looks like). If it is a "headliner change", still try to keep it reasonable. We always have a lot to cover.
3. In slight competition with point (2), don't omit interesting technical information when it is truly fun and engaging. A good chunk of our users are highly technical and enjoy learning how the sausage is made. Try to strike a balance between "terse and simple" and "nerdy details".
4. When relevant, briefly describe the problem being solved first, then describe the solution we chose. This contextualizes the change and gives the feature value and purpose.
5. When possible, provide visuals. They create interest / keep people hooked / break up the monotony.
6. Record images and videos at the default bevy resolution (1280x720)
7. Provide an accurate listing of authors that meaningfully contributed to the feature. Try to sort in order of "contribution scale". This is hard to define, but try to be fair. When in doubt, ask other contributors, SMEs, and/or maintainers.
8. Provide numbers and graphs where possible.  If something is faster, use numbers to back it up. We don't (yet) have automated graph generation in blog post style, so send data / info to me (@cart) if you want a graph made.

Use [Bevy's Dev Docs](https://dev-docs.bevyengine.orgl) to create links to symbols in your code:

```md
[`Camera3d`]: https://dev-docs.bevyengine.org/bevy/core_pipeline/core_3d/struct.Camera3d.html
```
When we publish the next release, we will update these links with their `docs.rs` equivalent.

## Headliners

Headliners are our "big ticket high importance / high profile" changes. They are listed briefly at the beginning of the blog post, their entries are roughly sorted "to the top", and they are given priority when it comes to "space in the blog post". If you think we missed something (or didn't prioritize something appropriately), let us know.

- Bevy Asset V2
- Android Suspend / Resume
- Deferred Renderer
- PCF Shadow Filtering
- Automatic Draw Batching and Instancing
- Sprite Instancing
- Renderer Performance Improvements
- One Shot Systems
- UI Node Outlines

## Sections

These are the sections we will cover in the blog post. If a section has been claimed, it will have `(claimed by X)` in the title. If it is unclaimed it will have `(unclaimed)` in the title. Let us know if we missed a section. We don't cover every feature, but we should cover pretty much everything that would be interesting to users. Note that what is interesting or challenging to implement is not necessarily something that is relevant to our blog post readers. And sometimes the reverse is true!

If you believe a section should be split up or reorganized, just bring it up here and we can discuss it.

- [x] Bevy Asset V2 (claimed by @cart)
  - [Bevy Asset V2][8624]
  - [Copy on Write AssetPaths][9729]
  - [Multiple Asset Sources](https://github.com/bevyengine/bevy/pull/9885)
- [x] Android Suspend / Resume (claimed by @mockersf)
  - [Android: handle suspend / resume][9937]
- [x] Deferred Renderer (claimed by @cart)
  - [Deferred Renderer][9258]
  - Potential blog author: @DGriffin91?
- [x] PCF Shadow Filtering: DirectionalLight and Spotlight (claimed by @cart)
  - [PCF For DirectionalLight/SpotLight Shadows][8006]
  - Potential blog author: @JMS55?
- [x] Automatic Draw Batching and Instancing (claimed by @superdump)
  - [Automatic batching/instancing of draw commands][9685]
- [x] GpuArrayBuffer and BatchedUniformBuffer (claimed by @superdump)
  - [Add GpuArrayBuffer and BatchedUniformBuffer][8204]
  - [Use GpuArrayBuffer for MeshUniform][9254]
  - [Reduce the size of MeshUniform to improve performance][9416]
- [x] Sprite Instancing (claimed by @superdump)
  - [Use instancing for sprites][9597]
- [x] Reorder Render Sets (claimed by @superdump)
  - [Reorder render sets, refactor bevy_sprite to take advantage][9236]
- [x] EntityHashMap (claimed by @superdump)
  - [Use EntityHashMap<Entity, T> for render world entity storage for better performance][9903]
- [ ] Other Renderer Performance Improvements (claimed by @cart)
  - [Directly copy data into uniform buffers][9865]
  - [Parallelize extract_meshes][9966]
  - Potential blog author: @james7132, @rparrett, or @superdump?
- [x] Import Second UV Map in glTF Files (claimed by @cart)
  - [Import the second UV map if present in glTF files.][9992]
  - Potential blog author: @pcwalton?
- [x] GLTF Material Emissive Strength (claimed by @cart)
  - [Add support for KHR_materials_emissive_strength][9553]
  - Potential blog author: @JMS55?
- [x] One Shot Systems (claimed by @Trashtalk217)
  - [One Shot Systems][8963]
  - [Add `RunSystem`][9366]
- [x] Disjoint Mutable World Access Via EntityMut (claimed by @cart)
  - [Allow disjoint mutable world access via `EntityMut`][9419]
  - Potential blog author: @JoJoJet?
- [x] Unified configure_sets API (claimed by @cart)
  - [Replace `IntoSystemSetConfig` with `IntoSystemSetConfigs`][9247]
  - Potential blog author: @geieredgar?
- [x] Ignore Component and Resource Ambiguities (claimed by @hymm)
  - [Ignore ambiguous components or resources][9895]
- [x] Wireframe Improvements (claimed by @IceSentry)
  - [Allow overriding global wireframe setting.][7328]
  - [Alternate wireframe override api][10023]
  - [use `Material` for wireframes][5314]
  - [Configurable Colors for Wireframes][5303]
  - Potential blog author: @IceSentry?
- [x] Simplify Parallel Iteration Methods (claimed by @cart)
  - [Simplify parallel iteration methods][8854]
  - Potential blog author: @JoJoJet?
- [x] UI Node Outlines (claimed by @cart)
  - [UI node outlines][9931]
  - Potential blog author: @ickshonpe?
- [x] Accessibility API Improvements (claimed by @ndarilek)
  - [Various accessibility API updates.][9989]
- [x] Pitch Audio Source (claimed by @mockersf)
  - [Added Pitch as an alternative sound source][9225]
  - Potential blog author: @basilefff?
- [x] Spatial Audio API Ergonomics (claimed by @rparrett)
  - [More ergonomic spatial audio][9800]
- [x] GamepadButtonInput Event (claimed by @cart)
  - [Add `GamepadButtonInput` event][9008]
  - Potential blog author: @bravely-beep?
- [x] TypePath Migraton and Improvements (claimed by @cart)
  - [reflect: `TypePath` part 2][8768]
- [x] AnimationPlayer API Improvements (claimed by @DevinLeamy)
  - [API updates to the AnimationPlayer][9002]
- [x] Improved Bevymark Example (claimed by @superdump)
  - [Enhance bevymark][9674]
- [ ] Improved Many Buttons Example (claimed by @cart)
  - [`many_buttons` enhancements][9712]
  - Potential blog author: @ickshonpe?
- [x] CI Improvements (claimed by @mockersf)
  - [Check for bevy_internal imports in CI][9612]
  - [run mobile tests on more devices / OS versions][9936]
- [x] ReflectBundle (claimed by @cart)
  - [bevy_scene: Add `ReflectBundle`][9165]
  - Potential blog author: @Shatur?
- [x] SceneInstanceReady Event (claimed by @cart)
  - [Add `SceneInstanceReady`][9313]
  - Potential blog author: @Shatur?
- [x] Limit Game Update Rate When Not Visible (claimed by @maniwani)
  - [Change `WinitPlugin` defaults to limit game update rate when window is not visible][7611]
- [x] Tracing Span Caching (claimed by @hymm)
  - [Cache System Tracing Spans][9390]
  - [Cache parallel iteration spans][9950]
  - Potential blog author: @hymm or @james7132?
- [x] Web Examples Improvements (claimed by @mockersf)
  - [example showcase - pagination and can build for WebGL2][9168]
  - [example showcase: switch default api to webgpu][9193]
- [x] Add system.map (claimed by @cart)
  - [Add `system.map(...)` for transforming the output of a system][8526]
  - Potential blog author: @JoJoJet
- [x] Split ComputedVisibility (claimed by @cart)
  - [Split `ComputedVisibility` into two components to allow for accurate change detection and speed up visibility propagation][9497]
  - Potential blog author: @JoJoJet
- [x] Get HSL Methods (claimed by @IDEDARY)
  - [Added methods to get HSL components from Color][9201]
- [x] Allow other plugins to create renderer resources (claimed by @cart)
  - [Allow other plugins to create renderer resources](https://github.com/bevyengine/bevy/pull/9925)
- [x] Insert and Remove Reflect Commands (claimed by @NoahShomette)
  - [implement insert and remove reflected entity commands][8895]
- [x] Unify `FixedTime` and `Time` (claimed by @maniwani)
  - [Unify FixedTime and Time](https://github.com/bevyengine/bevy/pull/8964)
- [x] StandardMaterial Extensions (claimed by @cart )
  - https://github.com/bevyengine/bevy/pull/7820
  - Potential blog author: @robtfm
- [x] Update Shader Imports (claimed by @cart)
  - Potential blog author: @robtfm
- [x] Bind Group Entry Ergonomics (claimed by @cart)
  - Potential blog author: @robtfm
- [x] Image sampler settings (claimed by @cart)
  - https://github.com/bevyengine/bevy/pull/9982
  - https://github.com/bevyengine/bevy/pull/10254

## Potential Sections

These are worth considering if we have time and space, but probably shouldn't take priority over the sections above. Very negotiable though this isn't an exact science!

- [Swap TransparentUi to use a stable sort][9598]
- [bevy_reflect: Opt-out attribute for `TypePath`][9140]
- Reflect path improvements
  - [Make the reflect path parser utf-8-unaware][9371]
  - [Make it so `ParsedPath` can be passed to GetPath][9373]
- Line Gizmo Fixes
  - [Fix gizmo draw order in 2D][9129]
  - [Fix gizmo line width issue when using perspective][9067]
  - [Fix gizmo lines deforming or disappearing when partially behind the camera][9470]
- [Allow Bevy to start from non-main threads on supported platforms][10020]
- Window Visibility
  - [Wait before making window visible][9692]
  - [User controlled window visibility][9355]
- [wgpu 0.17][9302]
- [invert face culling for negatively scaled gltf nodes][8859]
- [Fix wireframe for skinned/morphed meshes][9734]
- [Extend the default render range of 2D camera][9310]
- [Fix for vertical text bounds and alignment][9133]
- [Rename `Val` `evaluate` to `resolve` and implement viewport variant support][9568]
- [`impl From<String>` and `From<&str>` for `TextSection`][8856]
- [Have a separate implicit viewport node per root node + make viewport node `Display::Grid`][9637]
- [Store both the rounded and unrounded node size in Node][9923]
- [Round up for the batch size to improve par_iter performance][9814]
- [Replace AHash with a good sequence for entity AABB colors][9175]
- [Add panicking helpers for getting components from `Query`][9659]
- [Replaced `EntityCommand` Implementation for `FnOnce`][9604]
- [Move schedule name into `Schedule`][9600]
- [Refactor build_schedule and related errors][9579]
- [clean up configure_set(s) erroring][9577]
- [Only run event systems if they have tangible work to do][7728]

[5314]: https://github.com/bevyengine/bevy/pull/5314
[5303]: https://github.com/bevyengine/bevy/pull/5303
[7328]: https://github.com/bevyengine/bevy/pull/7328
[7611]: https://github.com/bevyengine/bevy/pull/7611
[7728]: https://github.com/bevyengine/bevy/pull/7728
[8006]: https://github.com/bevyengine/bevy/pull/8006
[8204]: https://github.com/bevyengine/bevy/pull/8204
[8526]: https://github.com/bevyengine/bevy/pull/8526
[8624]: https://github.com/bevyengine/bevy/pull/8624
[8768]: https://github.com/bevyengine/bevy/pull/8768
[8854]: https://github.com/bevyengine/bevy/pull/8854
[8856]: https://github.com/bevyengine/bevy/pull/8856
[8859]: https://github.com/bevyengine/bevy/pull/8859
[8895]: https://github.com/bevyengine/bevy/pull/8895
[8963]: https://github.com/bevyengine/bevy/pull/8963
[9002]: https://github.com/bevyengine/bevy/pull/9002
[9008]: https://github.com/bevyengine/bevy/pull/9008
[9067]: https://github.com/bevyengine/bevy/pull/9067
[9129]: https://github.com/bevyengine/bevy/pull/9129
[9133]: https://github.com/bevyengine/bevy/pull/9133
[9140]: https://github.com/bevyengine/bevy/pull/9140
[9154]: https://github.com/bevyengine/bevy/pull/9154
[9165]: https://github.com/bevyengine/bevy/pull/9165
[9168]: https://github.com/bevyengine/bevy/pull/9168
[9175]: https://github.com/bevyengine/bevy/pull/9175
[9193]: https://github.com/bevyengine/bevy/pull/9193
[9201]: https://github.com/bevyengine/bevy/pull/9201
[9225]: https://github.com/bevyengine/bevy/pull/9225
[9236]: https://github.com/bevyengine/bevy/pull/9236
[9247]: https://github.com/bevyengine/bevy/pull/9247
[9254]: https://github.com/bevyengine/bevy/pull/9254
[9258]: https://github.com/bevyengine/bevy/pull/9258
[9302]: https://github.com/bevyengine/bevy/pull/9302
[9310]: https://github.com/bevyengine/bevy/pull/9310
[9313]: https://github.com/bevyengine/bevy/pull/9313
[9355]: https://github.com/bevyengine/bevy/pull/9355
[9366]: https://github.com/bevyengine/bevy/pull/9366
[9371]: https://github.com/bevyengine/bevy/pull/9371
[9373]: https://github.com/bevyengine/bevy/pull/9373
[9390]: https://github.com/bevyengine/bevy/pull/9390
[9416]: https://github.com/bevyengine/bevy/pull/9416
[9419]: https://github.com/bevyengine/bevy/pull/9419
[9429]: https://github.com/bevyengine/bevy/pull/9429
[9453]: https://github.com/bevyengine/bevy/pull/9453
[9470]: https://github.com/bevyengine/bevy/pull/9470
[9497]: https://github.com/bevyengine/bevy/pull/9497
[9553]: https://github.com/bevyengine/bevy/pull/9553
[9568]: https://github.com/bevyengine/bevy/pull/9568
[9577]: https://github.com/bevyengine/bevy/pull/9577
[9579]: https://github.com/bevyengine/bevy/pull/9579
[9597]: https://github.com/bevyengine/bevy/pull/9597
[9598]: https://github.com/bevyengine/bevy/pull/9598
[9600]: https://github.com/bevyengine/bevy/pull/9600
[9604]: https://github.com/bevyengine/bevy/pull/9604
[9612]: https://github.com/bevyengine/bevy/pull/9612
[9630]: https://github.com/bevyengine/bevy/pull/9630
[9637]: https://github.com/bevyengine/bevy/pull/9637
[9659]: https://github.com/bevyengine/bevy/pull/9659
[9674]: https://github.com/bevyengine/bevy/pull/9674
[9685]: https://github.com/bevyengine/bevy/pull/9685
[9692]: https://github.com/bevyengine/bevy/pull/9692
[9712]: https://github.com/bevyengine/bevy/pull/9712
[9729]: https://github.com/bevyengine/bevy/pull/9729
[9734]: https://github.com/bevyengine/bevy/pull/9734
[9800]: https://github.com/bevyengine/bevy/pull/9800
[9814]: https://github.com/bevyengine/bevy/pull/9814
[9865]: https://github.com/bevyengine/bevy/pull/9865
[9882]: https://github.com/bevyengine/bevy/pull/9882
[9895]: https://github.com/bevyengine/bevy/pull/9895
[9903]: https://github.com/bevyengine/bevy/pull/9903
[9923]: https://github.com/bevyengine/bevy/pull/9923
[9931]: https://github.com/bevyengine/bevy/pull/9931
[9936]: https://github.com/bevyengine/bevy/pull/9936
[9937]: https://github.com/bevyengine/bevy/pull/9937
[9950]: https://github.com/bevyengine/bevy/pull/9950
[9966]: https://github.com/bevyengine/bevy/pull/9966
[9989]: https://github.com/bevyengine/bevy/pull/9989
[9992]: https://github.com/bevyengine/bevy/pull/9992
[10020]: https://github.com/bevyengine/bevy/pull/10020
[10023]: https://github.com/bevyengine/bevy/pull/10023

### CODE REVIEW

**Comment by hymm** [comment]
i can take "Ignore Component and Resource Ambiguities"

**Comment by JoJoJet** [comment]
IMO, https://github.com/bevyengine/bevy/pull/9630 and https://github.com/bevyengine/bevy/pull/8854 don't really need sections dedicated to them, whereas https://github.com/bevyengine/bevy/pull/8526 and https://github.com/bevyengine/bevy/pull/9497 probably should have sections (the latter was a major optimization for one of our slowest systems, which reduced the total frame time by 16% in one scene). I'd like to write the sections for these, but I won't claim them yet as I can't promise I'll get them done in time.

**Comment by rparrett** [comment]
https://github.com/bevyengine/bevy/pull/9882 was just addressing a regression introduced earlier in the cycle; probably don't need to mention it.

**Comment by MalekiRe** [comment]
I think it would be cool if the PR to make it so bevy_openxr crate can work would be of note, simply because now bevy can support vr games? I am not interested in claiming this though, just bringing it up in case it *IS* of enough general interest. I have a twitter video I can turn into a gif or a short clip so you can embed as well, of bevy vr functioning.

**Comment by IceSentry** [comment]
I can take all the wireframe stuff and I'll make a single entry for all of it.

**Comment by JMS55** [comment]
Assuming we can get them merged in time, transmission and TAA improvements should get a section. As of now I'm not committing to writing any sections due to lack of time sadly :/, but we'll see what happens as we get closer to next weekend.

**Comment by DevinLeamy** [comment]
I'll take the animation player update 👍

**Comment by superdump** [comment]
I can take all the batching/instancing and render optimisation stuff. I’ll try to weave a story. :)

**Comment by soqb** [comment]
i'm happy to take up the `TypePath` stuff.

**Comment by killercup** [comment]
I saw that https://github.com/bevyengine/bevy/pull/9154 does not work on docs.rs. As it works locally I'm gonna try and repro and fix before the next release.

If that works out I'm happy to write something about it! Otherwise let's keep quiet about it for now ;)

**Comment by robswain** [comment]
@cart I'm afraid you've tagged the wrong Rob Swain.

**Comment by Trashtalk217** [comment]
I can take the one shot systems part, but I feel like it should probably go together with the`RunSystem` part as they are so similar.

Of course I don't know how @Zeenobit feels about that.

**Comment by Zeenobit** [comment]
> I can take the one shot systems part, but I feel like it should probably go together with the`RunSystem` part as they are so similar.
> 
> Of course I don't know how @Zeenobit feels about that.

@Trashtalk217 I agree. I think the features are closely related. I have no issues with you taking over.
`run_system`/`run_system_with` should be fairly minor and straightforward. Let me know if you need any help there.

**Comment by IDEDARY** [comment]
Hi, can I take the `New get HSL methods on Color struct`? I'm the author of the PR and I would like to try contributing to the docs 😄

**Comment by cart** [comment]
> @robswain I'm afraid you've tagged the wrong Rob Swain.

Oops sorry! @superdump I tried using your discord username here :sweat_smile:

**Comment by cart** [comment]
> IMO, https://github.com/bevyengine/bevy/pull/9630 and https://github.com/bevyengine/bevy/pull/8854 don't really need sections dedicated to them, whereas https://github.com/bevyengine/bevy/pull/8526 and https://github.com/bevyengine/bevy/pull/9497

@JoJoJet I added the two sections. I left the parallelism ergonomics thing in because people love their ECS ergonomics. We can cut it later if it feels like we have too much noise.

> I think it would be cool if the PR to make it so bevy_openxr crate can work would be of note

@MalekiRe was that https://github.com/bevyengine/bevy/pull/8042?
If so, that happened in the 0.11 release cycle and we covered it in that blog post.

> Assuming we can get them merged in time, transmission and TAA improvements should get a section.

Agreed. I'll add things as they get merged.

> I can take all the batching/instancing and render optimisation stuff. I’ll try to weave a story. :)

@superdump would that include things like GpuArrayBuffer / reorder render sets / etc? Slightly more specific would be helpful so we don't step on each others' toes.

> I saw that https://github.com/bevyengine/bevy/pull/9154 does not work on docs.rs. As it works locally I'm gonna try and repro and fix before the next release.
> If that works out I'm happy to write something about it! Otherwise let's keep quiet about it for now ;)

@killercup ah good call. Thanks!

> I can take the one shot systems part, but I feel like it should probably go together with theRunSystem part as they are so similar.

@Trashtalk217 makes sense!

> Hi, can I take the New get HSL methods on Color struct? I'm the author of the PR and I would like to try contributing to the docs 😄

@IDEDARY sure!

**Comment by MalekiRe** [comment]
@cart no it was https://github.com/bevyengine/bevy/pull/9925

**Comment by cart** [comment]
@MalekiRe gotcha that makes sense. Added! If you could prepare a video clip of it working (mp4 format preferred) that would be much appreciated.

**Comment by mockersf** [comment]
I can take Android Suspend / Resume, and CI/Web examples improvements

**Comment by robtfm** [comment]
`Asset Loader Pre-Registration` - probably doesn't deserve a blog entry since it was in 0.11.1/0.11.2 and is largely invisible to end users?

**Comment by rparrett** [comment]
Happy to write something short about https://github.com/bevyengine/bevy/pull/9800 / spatial audio.

**Comment by cart** [comment]
> probably doesn't deserve a blog entry since it was in 0.11.1/0.11.2 and is largely invisible to end users?

We generally still cover relevant patch release content in the next major release blog as it hasn't been publicly covered yet (has only happened once I think). However the fact that it is largely invisible to users is a good point.  We can cut it.

**Comment by superdump** [comment]
I will claim:
* Automatic Draw Batching and Instancing
* GpuArrayBuffer and BatchedUniformBuffer
* Sprite Instancing
* Reorder Render Sets
* https://github.com/bevyengine/bevy/pull/9903
* Improved Bevymark Example

**Comment by nicopap** [comment]
IMO insert+remove commands are important. I often saw it as a feature request. If @NoahShomette is willing to write a section I think it should be included.

**Comment by NoahShomette** [comment]
> IMO insert+remove reflect commands are important. I often saw it as a feature request. If @NoahShomette is willing to write a section I think it should be included.

Yep, I'm more than happy to write it up!

**Comment by maniwani** [comment]
I'll claim "Limit Game Update Rate When Not Visible". Might be more digestible as "Limit Background FPS"?

**Comment by nakedible-p** [comment]
FWIW, I will be pretty busy this week, and can't promise I will have the time. I will know probably on 19th or 20th if I have time to do anything towards "Unify `FixedTime` and `Time`". If anyone else wants to claim it (@alice-i-cecile ?, @maniwani ?) I'm totally fine with it – but if no-one has done anything and I end up having the time, I can possibly write something at the last minute.

**Comment by Vrixyz** [comment]
I’d love to see:
- a reminder/teaser for the game jam (if it’s still planned)
- the source code used for any screenshots to reproduce/tweak more easily

**Comment by ndarilek** [comment]
Hey folks, sorry I didn't chime in earlier, $dayjob keeping me busy these past few months. :)

I knocked out a few paragraphs on the AccessKit integration improvements. If it's too late to get them in, that's fine and apologies for the duplicated effort. PR incoming shortly.

**Comment by hymm** [comment]
I wrote up something for "Tracing Span Caching"

**Comment by cart** [comment]
Claiming PCF Shadow Filtering

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 1]
```diff
@@ -0,0 +1,823 @@
...
+
+For those who don't know, Bevy is a refreshingly simple data-driven game engine built in Rust. You can check out our [Quick Start Guide](/learn/book/getting-started/) to try it today. It's free and open source forever! You can grab the full [source code](https://github.com/bevyengine/bevy) on GitHub. Check out [Bevy Assets](https://bevyengine.org/assets) for a collection of community-developed plugins, games, and learning resources.
+
+To update an existing Bevy App or Plugin to **Bevy 0.12**, check out our [0.11 to 0.12 Migration Guide](/learn/migration-guides/0.11-0.12/).
+
+Since our last release a few months ago we've added a _ton_ of new features, bug fixes, and quality of life tweaks, but here are some of the highlights:
+
+<!-- more -->
+
+## PCF Shadow Filtering
```
Comment (cart) [inline content/news/2023-10-21-bevy-0.12/index.md:18]: This needs to mention that its only for DirectionalLight and SpotLight

**Comment by mockersf** [comment]
Claiming Pitch Audio Source

**Comment by cart** [comment]
Claiming Deferred Rendering

**Comment by cart** [comment]
Claiming Update Shader Imports

**Comment by cart** [comment]
@soqb still planning on doing the TypePath section or should I pick that up?

**Comment by cart** [comment]
Claiming StandardMaterial Extensions

**Comment by cart** [comment]
Claiming ImageSamplerSettings

**Comment by cart** [comment]
Claiming bind group ergonomics

**Comment by cart** [comment]
Claiming Disjoint Mutable World Access Via EntityMut, UI Node Borders, and Unified configure_sets API

**Comment by soqb** [comment]
> @soqb still planning on doing the TypePath section or should I pick that up?

i did end up writing something a while ago but i never felt like it was in a usable state. feel free to pick it up since i don't think i've got time for a good few days.

i'm sorry i wasn't more transparent about all this.

**Comment by cart** [comment]
Claiming "everything else"

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 2]
```diff
@@ -0,0 +1,1638 @@
...
+
+Avoiding rebinding is both a big performance benefit for CPU-driven rendering and is necessary to enable GPU-driven rendering.
+
+To avoid rebinds, the general data pattern we are aiming for is:
+
+* For each data type (meshes, materials, transforms, textures), create a single array (or a small number of arrays) containing all of the items of that data type
+* Bind these arrays a small number of times (ideally once), avoiding per-entity/per-draw rebinds
+
+In **Bevy 0.12** we've started this process in earnest! We've made a number of architectural changes that are already yielding fruit. Thanks to these changes, we can now [automatically batch and instance draws](#automatic-batching-instancing-of-draw-commands) for entities with the exact same mesh and material. And as we progress further down this path, we can batch/instance across a wider variety of cases, cutting out more and more CPU work until eventually we are "fully GPU-driven".
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:1449]: This is the broken link.

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 3]
```diff
@@ -0,0 +1,1638 @@
...
+    commands.spawn(PitchBundle {
+        source: pitch_handle,
+        ..default()
+    });
+}
+```
+
+Audio is generated at the given frequency using a [sine wave](https://en.wikipedia.org/wiki/Sine_wave#Audio_example). More complex sounds can be created by playing several pitch audio sources at the same time, like chords or hamonics.
+
+## Automatic Batching/Instancing of Draw Commands
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:1380]: This is what we need to link to.

File: content/news/2023-10-21-bevy-0.12/index.md (hunk 2 above)
Comment (rparrett) [reply content/news/2023-10-21-bevy-0.12/index.md:1449]: I **think** that markdownlint is expecting 
`#automatic-batchinginstancing-of-draw-commands`, and that the different in markdownlint and zola's slugification could be worked around by adding spaces around the `/` in that heading.

Comment (cart) [reply content/news/2023-10-21-bevy-0.12/index.md:1449]: Yeah this isn't actually a broken link in Zola. Its a markdownlint behavior mismatch.

File: content/news/2023-10-21-bevy-0.12/index.md (hunk 3 above)
Comment (cart) [reply content/news/2023-10-21-bevy-0.12/index.md:1380]: Resolving as we already link to this. We can use the comment to resolve the markdownlint behavior mismatch.

File: content/news/2023-10-21-bevy-0.12/index.md (hunk 2 above)
Comment (rparrett) [reply content/news/2023-10-21-bevy-0.12/index.md:1449]: There's still a mismatch with `(space)/(space)`, but replacing with `and` or explicitly defining the anchor seems to work:

```
## Automatic Batching And Instancing of Draw Commands {#automatic-batching-instancing-of-draw-commands}
```

File: content/news/2023-11-04-bevy-0.12/index.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
 * **Renderer Optimizations**: Bevy's renderer dataflow has been reworked to squeeze out more performance and prepare the way for future GPU-driven rendering.
 * **One Shot Systems**: ECS Systems can now be run on-demand from other systems!
 
-<!-- more -->
```
Comment (rparrett) [inline content/news/2023-11-04-bevy-0.12/index.md]: Oops. Two more of these below.

File: content/news/2023-11-04-bevy-0.12/index.md [hunk 5]
```diff
@@ -0,0 +1,3050 @@
...
+
+## The Road to GPU-driven Rendering
+
+<div class="release-feature-authors">authors: @superdump (Rob Swain), @james-j-obrien, @JMS55, @inodentry, @robtfm, @nicopap, @teoxoy, @IceSentry, @Elabajaba</div>
+
+Bevy's renderer performance for 2D and 3D meshes can improve a lot. There are bottlenecks on both the CPU and GPU side, which can be lessened to give significantly higher frame rates. As always with Bevy, we want to make the most of the platforms you use, from the constraints of WebGL2 and mobile devices, to the highest-end native discrete graphics cards. A solid foundation can support all of this.
+
+In **Bevy 0.12** we have started reworking rendering data structures, data flow, and draw patterns to unlock new optimizations. This enabled the **Automatic Batching and Instancing** we landed in **Bevy 0.12** and also helps pave the way for other significant wins in the future, such as GPU-driven rendering. We aren't quite ready for GPU-driven rendering, but we've started down that path in **Bevy 0.12**!
+
+### What are CPU* and GPU-driven rendering?
```
Comment (rparrett) [inline content/news/2023-11-04-bevy-0.12/index.md:714]:
```suggestion
### What are CPU- and GPU-driven rendering?
```

File: content/news/2023-11-04-bevy-0.12/index.md [hunk 6]
```diff
@@ -0,0 +1,3050 @@
...
+
+The example showcase tool can now build all examples for WebGL2 or WebGPU. This is used to update
+the website with all Wasm-compatible examples, which you can find
+[here](https://bevyengine.org/examples/) for WebGL2, or
+[here](https://bevyengine.org/examples-webgpu/) for WebGPU.
+
+It is now also capable of capturing a screenshot while running all examples:
+
+```sh
+cargo run -p example-showcase -* run --screenshot
```
Comment (rparrett) [inline content/news/2023-11-04-bevy-0.12/index.md:1635]:
```suggestion
cargo run -p example-showcase -- run --screenshot
```

**Comment by mockersf** [comment]
last minute https://github.com/bevyengine/bevy/pull/9506 should be there

**Comment by asafigan** [comment]
I think it would be helpful to clarify if One Shot systems (`SystemId`s) need to be manually removed or not. Do they act like assets and `Handle`s where they are cleaned up when there are no more references to them? Should you run `remove_system` to prevent memory leaks?

File: content/news/2023-11-04-bevy-0.12/index.md [hunk 7]
```diff
@@ -0,0 +1,3116 @@
...
+// Later in your app, spawn the UI node with your material!
+commands.spawn(MaterialNodeBundle {
+    style: Style {
+        position_type: PositionType::Absolute,
+        width: Val::Px(250.0),
+        height: Val::Px(250.0),
+        ..default()
+    },
+    material: materials.add(CircleMaterial {
+        color: Color::WHITE.into(),
```
Comment (mahulst) [inline content/news/2023-11-04-bevy-0.12/index.md:1248]: Not entirely sure if this is a bug or something the shader does, but why is the circle green if the color input is white?

Comment (mockersf) [reply content/news/2023-11-04-bevy-0.12/index.md:1248]: it should be white with just this extract. the full example has a system in update that changes the color every frame

File: content/news/2023-11-04-bevy-0.12/index.md [hunk 8]
```diff
@@ -0,0 +1,3116 @@
...
+The [`StandardMaterial`] now supports a number of light transmission-related properties:
+
+* `specular_transmission`
+* `diffuse_transmission`
+* `thickness`
+* `ior`
+* `attenuation_color`
+* `attenuation_distance`
+
+These allow you to more realistically represent a wide variety of physical materials, including **clear and frosted glass, water, plastic, foliage, paper, wax, marble, porcelain and more**.
```
Comment (alice-i-cecile) [inline content/news/2023-11-04-bevy-0.12/index.md:142]:
```suggestion
These allow you to more realistically represent a wide variety of physical materials, including **clear and frosted glass, water, plastic, foliage, paper, wax, marble, porcelain and more**.
To see what this looks like for real time rendering in Bevy 0.12, check out this [demo video](https://www.youtube.com/watch?v=t1XdxZKZ-us).
```

**Comment by alice-i-cecile** [comment]
> I think it would be helpful to clarify if One Shot systems (`SystemId`s) need to be manually removed or not. Do they act like assets and `Handle`s where they are cleaned up when there are no more references to them? Should you run `remove_system` to prevent memory leaks?

I think this belongs in the docs, not this post. But yeah, there's no reference counting or anything.

File: content/news/2023-11-04-bevy-0.12/index.md (hunk 8 above)
Comment (cart) [reply content/news/2023-11-04-bevy-0.12/index.md:142]: Oops pushed from my branch before checking comments. Just added something similar!

**Pull request merged**